src/components/definition-list/definition-list.style.ts

```diff
@@ -39,49 +29,42 @@ StyledDl.defaultProps = {
   theme: baseTheme,
 };
 
-export interface StyledDtDivProps {
-  /** This string will specify the text align styling of the `<dt></dt>`. */
-  dtTextAlign?: ElementAlignment;
-}
-
-export const StyledDtDiv = styled.div<StyledDtDivProps>`
-  ${space}
-  ${({ dtTextAlign }) => css`
-    text-align: ${dtTextAlign};
-  `}
-`;
-
-StyledDtDiv.defaultProps = {
-  theme: baseTheme,
-};
-
-export interface StyledDdDivProps {
-  /** This string will specify the text align styling of the `<dd></dd>`. */
-  ddTextAlign?: ElementAlignment;
-}
-
-export const StyledDdDiv = styled.div<StyledDdDivProps>`
-  ${({ ddTextAlign }) => css`
-    text-align: ${ddTextAlign};
-  `}
-`;
-
-export const StyledDt = styled.dt`
+export const StyledDt = styled.dt<
+  Pick<DlProps, "asSingleColumn" | "dtTextAlign">
+>`
   ${space}
   font-size: var(--fontSizes100);
   font-weight: 700;
   color: var(--colorsUtilityYin090);
+  ${({ asSingleColumn }) =>
+    asSingleColumn &&
+    css`
+      grid-column: 1;
```

Comment:
**comment**: I'm a bit confused by this `grid-column` property - both here and on line 63. They're only applied when `asSingleColumn` is true, but that's precisely the case (line 13 of this same file) that `display: grid;` is *not* applies, which makes it do nothing. See screenshot from CSS inspector:
![image](https://github.com/Sage/carbon/assets/25186376/07fdc413-42da-4d0e-92f1-e085d856c7cb)

I'm not sure why any of these changes are necessary - things look visually OK to me (although I've only done a quick side-by-side comparison with master) even with this clearly not working, and since we're not trying to produce visual changes I don't see the need for any new CSS (the only change to styles in this PR might be where various props come from).

Reply:
Yep I have the logic inverted 🤦 they should be applied when the grid layout is set not when it is a single column